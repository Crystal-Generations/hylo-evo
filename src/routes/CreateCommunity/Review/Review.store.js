export const MODULE_NAME = `Review`
export const CREATE_COMMUNITY = `${MODULE_NAME}/CREATE_COMMUNITY`

export function createCommunity (name, slug) {
  return {
    type: CREATE_COMMUNITY,
    graphql: {
      query: `mutation ($data: CommunityInput) {
        createCommunity(data: $data) {
          community {
            name
            slug
          }
        }
      }
      `,
      variables: {
        data: {
          name,
          slug
        }
      }
    },
    meta: {
      optimistic: false,
      extractModel: 'Membership'
    }
  }
}
