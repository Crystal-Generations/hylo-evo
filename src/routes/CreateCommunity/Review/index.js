import React, { Component } from 'react'
import '../CreateCommunity.scss'
import LeftSidebar from '../../Signup/LeftSidebar'
import TextInput from 'components/TextInput'
import { hyloNameWhiteBackground } from 'util/assets'
import { bgImageStyle } from 'util/index'
import ModalFooter from '../ModalFooter'

const theme = {
  inputStyle: 'modal-input',
  wrapperStyle: 'center'
}
export default class Review extends Component {
  setState = () => {

  }
  render () {
    return <div styleName='flex-wrapper'>
      <LeftSidebar
        header='Everything looking good?'
        body='You can always come back and change your details at any time'
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 4/4</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-vertically'>
          <TextInput
            type='text'
            name='community-name'
            onChange={this.setState('community-name')}
            inputRef={input => { this.email = input }}
            theme={theme}
            placeholder='Choose a domain name'
          />
        </div>
      </div>
      <ModalFooter
        submit={this.submit}
        previous={this.previous}
        showPrevious={false}
        continueText={'Finish Up'}
        />
    </div>
  }
}
