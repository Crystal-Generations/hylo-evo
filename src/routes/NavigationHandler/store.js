const NAVIGATE = 'NAVIGATE'
const RESET_NAVIGATION = 'RESET_NAVIGATION'

export function navigate (to) {
  return {
    type: NAVIGATE,
    payload: {to}
  }
}

export function resetNavigation (to) {
  return {type: RESET_NAVIGATION}
}

const reducer = (state = null, { type, error, payload, meta }) => {
  switch (type) {
    case NAVIGATE:
      return payload.to
    case RESET_NAVIGATION:
      return null
  }
  return state
}

export default reducer
