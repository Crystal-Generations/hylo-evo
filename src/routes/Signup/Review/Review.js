import React, { Component } from 'react'
import { map } from 'lodash'
import cx from 'classnames'
import LeftSidebar from '../LeftSidebar'
import SignupModalFooter from '../SignupModalFooter'
import UploadImageSection from '../UploadImageSection'
import Pillbox from 'components/Pillbox'

import '../Signup.scss'

export default class AddLocation extends Component {
  constructor () {
    super()
    this.state = {
      edits: {
        name: null,
        email: null,
        password: null,
        location: null
      },
      readOnly: {
        name: true,
        email: true,
        password: true,
        location: true
      }
    }
  }
  handleInputChange = (event, name) => {
    const value = event.target.value
    this.setState({
      edits: {
        ...this.state.edits,
        [name]: value
      }
    })
  }

  makeEditable = (event, name) => {
    this.setState({
      readOnly: {
        ...this.state.edits,
        [name]: false
      }
    })
  }
  submit = () => {
    const location = this.state.location
    this.props.updateUserSettings({location})
    this.props.goToNextStep()
  }

  previous = () => {
    this.props.goToPreviousStep()
  }

  updateSettingDirectly = (key, setChanged) => value => {
    const { edits, changed } = this.state
    // setChanged && setConfirm('You have unsaved changes, are you sure you want to leave?')
    this.setState({
      changed: setChanged ? true : changed,
      edits: {
        ...edits,
        [key]: value
      }
    })
  }

  componentWillMount = () => {
    const { currentUser } = this.props
    if (currentUser && currentUser.settings && currentUser.settings.signupInProgress === 'false') this.props.goBack()
  }

  componentDidMount = () => {
    this.props.fetchMySkills()
  }

  render () {
    const currentAvatarUrl = this.state.edits.avatarUrl

    const { currentUser, uploadImagePending } = this.props
    const skills = currentUser && currentUser.skills && currentUser.skills.toRefArray()
    return <div styleName='wrapper'>
      <LeftSidebar
        header='Everything looking good?'
        body='You can always come back and change your details at any time.'
      />
      <div styleName='detail'>
        <span styleName='white-text step-count'>STEP 4/4</span>
        <br />
        <div styleName='center'>
          {currentUser && <UploadImageSection
            avatarUrl={currentAvatarUrl}
            updateSettingDirectly={this.updateSettingDirectly}
            currentUser={currentUser}
            loading={uploadImagePending}
          />}
        </div>
        <div styleName='three-column-input'>
          <div styleName='column-left'>YOUR NAME</div>
          <div styleName='column-center'>
            <input
              styleName='signup-input'
              onChange={(e) => this.handleInputChange(e, 'name')}
              onKeyPress={event => {
                if (event.key === 'Enter') {
                  this.submit()
                  this.props.goToNextStep()
                }
              }}
              autoFocus
              value={this.state.edits.name || currentUser && currentUser.name}
              readOnly={this.state.readOnly.name}
            />
          </div>
          <div styleName='column-right'>
            <span styleName='edit-button' onClick={() => this.makeEditable('name')}>Edit</span>
          </div>
        </div>
        <div styleName='three-column-input'>
          <div styleName='column-left'>YOUR EMAIL</div>
          <div styleName='column-center'>
            <input
              styleName='signup-input'
              onChange={(e) => this.handleInputChange(e, 'email')}
              onKeyPress={event => {
                if (event.key === 'Enter') {
                  this.submit()
                  this.props.goToNextStep()
                }
              }}
              autoFocus
              value={this.state.edits.email || currentUser && currentUser.email}
              readOnly={this.state.readOnly.name}
            />
          </div>
          <div styleName='column-right'>
            <span styleName='edit-button' onClick={() => this.makeEditable('email')}>Edit</span>
          </div>
        </div>
        <div styleName='three-column-input'>
          <div styleName='column-left'>LOCATION</div>
          <div styleName='column-center'>
            <input
              styleName='signup-input'
              onChange={(e) => this.handleInputChange(e, 'location')}
              onKeyPress={event => {
                if (event.key === 'Enter') {
                  this.submit()
                  this.props.goToNextStep()
                }
              }}
              autoFocus
              value={this.state.edits.location || currentUser && currentUser.location}
              readOnly={this.state.readOnly.name}
            />
          </div>
          <div styleName='column-right'>
            <span styleName='edit-button' onClick={() => this.makeEditable('location')}>Edit</span>
          </div>
        </div>
        <div styleName='three-column-input'>
          <div styleName='column-left'>SKILLS</div>
          <div styleName='column-center'>
            <div
              styleName={cx('pill-container', {expanded, collapsed: !expanded})}>
              <Pillbox
                pills={map(skills, skill => ({...skill, label: skill.name}))}
                handleInputChange={this.handleInputChange}
                handleAddition={this.handleAddition}
                handleDelete={this.handleDelete}
                editable={false}
                addLabel='Add a Skill'
                placeholder={null}
                suggestions={null}
              />
            </div>
          </div>
          <div styleName='column-right'>
            <span styleName='edit-button'>Edit</span>
          </div>
        </div>
        <div>
          <SignupModalFooter submit={this.submit} previous={this.previous} showPrevious={false} />
        </div>
      </div>
    </div>
  }
}

export function Pill ({skill}) {
  return <span styleName='skill'>
    {skill.name}
  </span>
}
