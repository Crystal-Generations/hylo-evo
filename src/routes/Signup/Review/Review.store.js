export const FETCH_MY_SKILLS = `FETCH_MY_SKILLS`
export const UPDATE_USER_SETTINGS = `UPDATE_USER_SETTINGS`

export function fetchMySkills (limit = 20) {
  return {
    type: FETCH_MY_SKILLS,
    graphql: {
      query: `query ($limit: Int) {
        me {
          id
          skills (first: $limit) {
            items {
              id
              name
            }
          }
        }
      }`,
      variables: { limit }
    },
    meta: {
      extractModel: 'Me'
    }
  }
}
