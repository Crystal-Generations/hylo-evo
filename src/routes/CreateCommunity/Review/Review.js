import React, { Component } from 'react'
import '../CreateCommunity.scss'
import ModalSidebar from 'components/ModalSidebar'
import TextInput from 'components/TextInput'
import { hyloNameWhiteBackground, groovingAxolotl } from 'util/assets'
import { bgImageStyle } from 'util/index'
import ModalFooter from 'components/ModalFooter'
import { find } from 'lodash'
import { get } from 'lodash/fp'
import {
  slugValidatorRegex,
  formatDomainWithUrl,
  removeUrlFromDomain,
  invalidSlugMessage
} from '../util'

export default class Review extends Component {
  constructor () {
    super()

    this.state = {
      readOnly: {
        name: true,
        domain: true
      },
      edits: {
        name: '',
        domain: '',
        defaultTopics: [],
        templateName: null,
        privacy: null,
        changed: false
      }
    }
  }

  editHandler = (name) => {
    this.setState({
      readOnly: {
        ...this.state.readOnly,
        [name]: false
      }
    })
    this[name].select()
  }

  handleInputChange = (event, name) => {
    let value = event.target.value
    if (name === 'domain') {
      value = removeUrlFromDomain(value)
      if (value !== '') {
        this.props.fetchCommunityExists(value)
      }
    }
    this.setState({
      edits: {
        ...this.state.edits,
        [name]: value,
        changed: true
      }
    })
  }

  onEnter = event => {
    if (event.key === 'Enter') {
      this.errorCheckAndSubmit()
    }
  }

  submit = () => {
    const { name, domain, defaultTopics, template, networkId } = this.state.edits
    this.props.createCommunity(
      // privacy,
      name,
      domain,
      template ? template.id : null,
      defaultTopics,
      networkId
    )
      .then(({ error }) => {
        if (error) {
          this.setState({
            error: 'There was an error, please try again.'
          })
        } else {
          this.props.goToCommunity(`${domain}`)
        }
      })
  }

  errorCheckAndSubmit = () => {
    const { name, domain } = this.state.edits
    if (name === '' || domain === '') {
      this.setState({
        error: 'Please fill in each field.'
      })
    } else if (this.props.domainExists) {
      this.setState({
        error: 'This URL already exists. Try another.'
      })
    } else if (!slugValidatorRegex.test(removeUrlFromDomain(domain))) {
      formatDomainWithUrl(domain)
      this.setState({
        error: invalidSlugMessage
      })
    } else {
      this.submit()
    }
  }

  componentWillMount = () => {
    const { privacy } = this.props
    const privacyOption = find(privacyOptions, { label: privacy })
    const selectedprivacy = get('label', privacyOption) // set to Private by default
    this.setState({
      edits: {
        name: get('name', this.props) || '',
        domain: get('domain', this.props) || '',
        defaultTopics: get('defaultTopics', this.props) || [],
        networkId: get('networkId', this.props) || '',
        privacy: selectedprivacy,
        template: get('template', this.props) || null
      }
    })
  }

  formatDomainWithUrl (domain) {
    if (!domain) return null
    let formattedDomain = domain.replace('hylo.com/c/', '').replace('hylo.com/c', '')
    if (formattedDomain !== '') {
      formattedDomain = 'hylo.com/c/' + formattedDomain
    }
    return formattedDomain
  }

  removeUrlFromDomain (domain) {
    return domain.replace('hylo.com/c/', '')
  }

  render () {
    const { networkName } = this.props

    return <div styleName='flex-wrapper'>
      <ModalSidebar
        imageUrl={groovingAxolotl}
        onClick={this.props.goHome}
        theme={sidebarTheme}
        header='Everything looking good?'
        body='You can always come back and change your details at any time'
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 4/4</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-review'>
          <ReviewTextInput
            label={'Community Name'}
            value={this.state.edits.name || ''}
            readOnly={this.state.readOnly.name}
            editHandler={() => this.editHandler('name')}
            onEnter={this.onEnter}
            onChange={(e) => this.handleInputChange(e, 'name')}
            inputRef={(input) => { this.name = input }}
          />
          <ReviewTextInput
            label={'URL'}
            value={formatDomainWithUrl(this.state.edits.domain) || ''}
            readOnly={this.state.readOnly.domain}
            editHandler={() => this.editHandler('domain')}
            onEnter={this.onEnter}
            onChange={(e) => this.handleInputChange(e, 'domain')}
            inputRef={(input) => { this.domain = input }}
          />
          <ReviewTextInput
            label={'Community Template'}
            value={this.state.edits.template ? this.state.edits.template.displayName : ''}
            readOnly
            editHandler={() => this.props.goToStep('template')}
          />
          <ReviewTextInput
            label={'Default Topics'}
            value={this.state.edits.defaultTopics}
            readOnly
            editHandler={() => this.props.goToStep('template')}
          />
          {networkName && <ReviewTextInput
            label={'Network'}
            value={networkName}
            readOnly
          />}
          {/* <ReviewTextInput */}
          {/* label={'Privacy'} */}
          {/* value={this.state.edits.privacy} */}
          {/* onEnter={this.onEnter} */}
          {/* editHandler={() => this.props.goToPrivacyStep()} */}
          {/* /> */}

        </div>
        { this.state.error && <span styleName='review-arrow-up' /> }
        { this.state.error && <span styleName='review-error'>{this.state.error}</span>}
      </div>
      <ModalFooter
        submit={this.errorCheckAndSubmit}
        previous={this.previous}
        showPrevious={false}
        continueText={'Finish Up'}
      />
    </div>
  }
}

export function ReviewTextInput ({ label, value, editHandler, onChange, readOnly = true, inputRef }) {
  return <div styleName='review-input-text-row'>
    <div styleName='review-input-text-label'>
      <span>{label}</span>
    </div>
    <div styleName='review-input-text'>
      { Array.isArray(value)
        ? <div styleName='reviewPills'>{value.map((v, index) => <Pill key={index} value={v} />)}</div>
        : <TextInput
          type='text'
          value={value}
          theme={inputTheme}
          readOnly={readOnly}
          noClearButton
          onChange={onChange}
          inputRef={inputRef}
        />
      }
    </div>
    {editHandler && <div styleName='review-input-edit'>
      <span styleName='edit-button' onClick={editHandler}>Edit</span>
    </div>}
  </div>
}

function Pill ({ value }) {
  return <span styleName='defaultTopic'>
    {value}
  </span>
}

const inputTheme = {
  inputStyle: 'modal-input partial',
  wrapperStyle: 'center'
}

const sidebarTheme = {
  sidebarHeader: 'sidebar-header-full-page',
  sidebarText: 'gray-text sidebar-text-full-page'
}

const privacyOptions = [
  { id: '0', label: 'public' },
  { id: '1', label: 'private' },
  { id: '2', label: 'unlisted' }
]
