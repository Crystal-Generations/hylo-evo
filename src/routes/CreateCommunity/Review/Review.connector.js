import { get } from 'lodash/fp'
import { connect } from 'react-redux'
import { push } from 'connected-react-router'
import { addCommunityName, addCommunityDomain, fetchCommunityExists } from '../CreateCommunity.store'
import { createCommunity, getNetwork } from './Review.store'

export function mapStateToProps (state, props) {
  const networkId = get('networkId', state.CreateCommunity)
  const network = getNetwork(state, { networkId })
  const templateId = get('templateId', state.CreateCommunity)
  const template = get('communityTemplates', state.CreateCommunity).find(ct => ct.id === templateId)

  return {
    defaultTopics: get('defaultTopics', state.CreateCommunity),
    domain: get('domain', state.CreateCommunity),
    domainExists: get('domainExists', state.CreateCommunity),
    name: get('name', state.CreateCommunity),
    networkId,
    networkName: get('name', network),
    privacy: get('privacy', state.CreateCommunity),
    template
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    goToCommunity: (communityDomain) => dispatch(push(`/c/${communityDomain}`)),
    clearNameFromCreateCommunity: () => dispatch(addCommunityName(null)),
    clearDomainFromCreateCommunity: () => dispatch(addCommunityDomain(null)),
    createCommunity: (name, slug, defaultTopics, networkId) => dispatch(createCommunity(name, slug, defaultTopics, networkId)),
    goToStep: (step) => dispatch(push('/create-community/' + step)),
    goHome: () => dispatch(push('/')),
    fetchCommunityExists: (slug) => dispatch(fetchCommunityExists(slug))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
