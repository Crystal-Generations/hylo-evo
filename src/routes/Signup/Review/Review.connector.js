import { connect } from 'react-redux'
import { push, goBack } from 'react-router-redux'
import getMe from 'store/selectors/getMe'
import { UPLOAD_IMAGE } from 'store/constants'
import { fetchMySkills } from './Review.store'
import { updateUserSettings } from 'store/actions/updateUserSettings'

export function mapStateToProps (state, props) {
  const uploadImagePending = state.pending[UPLOAD_IMAGE]
  return {
    currentUser: getMe(state),
    uploadImagePending
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    updateUserSettings: (changes) => dispatch(updateUserSettings(changes)),
    goToNextStep: () => dispatch(push('/')),
    goToPreviousStep: () => dispatch(push('/signup/add-skills')),
    goBack: () => dispatch(goBack()),
    fetchMySkills: () => dispatch(fetchMySkills())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
