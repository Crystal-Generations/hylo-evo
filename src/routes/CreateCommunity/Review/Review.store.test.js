import { createCommunity } from './Review.store'

describe('createCommunity', () => {
  describe('without networkId', () => {
    it('omits networkId from variables', () => {
      const action = createCommunity('thename', 'theslug')
      expect(action).toMatchSnapshot()
      expect(action.graphql.variables.data.networkId).toEqual(undefined)
    })
  })

  describe('with template, topics and networkId', () => {
    it('includes networkId in variables', () => {
      const networkId = 123
      const templateId = 1
      const topics = ['topic']
      const action = createCommunity('thename', 'theslug', templateId, topics, networkId)
      expect(action).toMatchSnapshot()
      expect(action.graphql.variables.data.networkId).toEqual(networkId)
      expect(action.graphql.variables.data.communityTemplateId).toEqual(templateId)
      expect(action.graphql.variables.data.defaultTopics).toEqual(topics)
    })
  })
})
