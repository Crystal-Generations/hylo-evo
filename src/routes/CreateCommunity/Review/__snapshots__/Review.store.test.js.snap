// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createCommunity with template, topics and networkId includes networkId in variables 1`] = `
Object {
  "graphql": Object {
    "query": "mutation ($data: CommunityInput) {
        createCommunity(data: $data) {
          id
          hasModeratorRole
          community {
            id
            name
            slug
            network {
              id
            }
          }
        }
      }
      ",
    "variables": Object {
      "data": Object {
        "communityTemplateId": 1,
        "defaultTopics": Array [
          "topic",
        ],
        "name": "thename",
        "networkId": 123,
        "slug": "theslug",
      },
    },
  },
  "meta": Object {
    "analytics": "Community Created",
    "communityTemplateId": 1,
    "defaultTopics": Array [
      "topic",
    ],
    "extractModel": "Membership",
    "name": "thename",
    "networkId": 123,
    "slug": "theslug",
  },
  "type": "Review/CREATE_COMMUNITY",
}
`;

exports[`createCommunity without networkId omits networkId from variables 1`] = `
Object {
  "graphql": Object {
    "query": "mutation ($data: CommunityInput) {
        createCommunity(data: $data) {
          id
          hasModeratorRole
          community {
            id
            name
            slug
            network {
              id
            }
          }
        }
      }
      ",
    "variables": Object {
      "data": Object {
        "communityTemplateId": undefined,
        "defaultTopics": undefined,
        "name": "thename",
        "slug": "theslug",
      },
    },
  },
  "meta": Object {
    "analytics": "Community Created",
    "communityTemplateId": undefined,
    "defaultTopics": undefined,
    "extractModel": "Membership",
    "name": "thename",
    "slug": "theslug",
  },
  "type": "Review/CREATE_COMMUNITY",
}
`;
